docs/ref/templates/builtins.txt

```diff
@@ -961,40 +961,47 @@ output (as a string) inside a variable. This is useful if you want to use
 .. versionadded:: 5.1
 
 Outputs the query string from the given :class:`~django.http.QueryDict`
-instance. If the ``QueryDict`` is empty, then the output will be an empty
-string. Otherwise, the query string will be displayed with a leading "?".
+instance, or ``request.GET`` if not given and the
+``django.template.context_processors.request`` context processor is enabled.
+If the ``QueryDict`` is empty, then the output will be an empty string.
+Otherwise, the query string will be displayed with a leading "?".
 
 The following example outputs the current query string verbatim. So if the
 query string is ``?color=green&=size=M``, the output would be
 ``?color=green&size=M``:
 
 .. code-block:: html+django
 
-    {% query_string request.GET %}
+    {% query_string %}
+
+Using a custom ``QueryDict``:
+
+.. code-block:: html+django
+
+    {% query_string my_query_dict %}
```

Comment:
Passing a custom query dict complicates this behavior, and this example doesn't explains anything.

Reply:
Do you have any suggestions here? Carlton had an example in the forum thread, though I don't know if you're asking for an example here or just more explanation, but I'm at a bit of a loss as to what more I could explain.

Reply:
Okay, now I understand, I've updated it.